agent/engine/docker_task_engine.go

```diff
@@ -1854,32 +1854,31 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 		}
 	}
 
-	// This is a short term solution only for specific regions until AWS SDK Go is upgraded to V2
+	// We're currently relying on AWS SDK Go V1 to obtain/format the correct cloudwatch endpoint.
+	// This is will need to be updated once we upgrade to AWS SDK Go V2.
 	if hostConfig.LogConfig.Type == logDriverTypeAwslogs {
-		region := engine.cfg.AWSRegion
-		if region == "eu-isoe-west-1" || region == "us-isof-south-1" || region == "us-isof-east-1" {
-			endpoint := ""
-			dnsSuffix := ""
-			partition, ok := ep.PartitionForRegion(ep.DefaultPartitions(), region)
-			if !ok {
-				logger.Warn("No partition resolved for region. Using AWS default", logger.Fields{
-					"region":           region,
-					"defaultDNSSuffix": ep.AwsPartition().DNSSuffix(),
-				})
-				dnsSuffix = ep.AwsPartition().DNSSuffix()
+		region := hostConfig.LogConfig.Config["awslogs-region"]
```

Comment:
Would `hostConfig.LogConfig.Config["awslogs-region"]` ever be empty or null? If yes, should fallback to use `region := engine.cfg.AWSRegion` before moving to line 1863?

Reply:
RIght so `awslogs-region` shouldn't be empty/null ever. If a customer were to omit this option from the task definition then it would be considered invalid.

Reply:
My original thought process was to also default to `engine.cfg.AWSRegion` but after testing manually I couldn't find a scenario where this is not aparent from the task payload. Although I'm open to have this as a safety net or maybe fail the task.